velox/functions/sparksql/tests/SparkCastExprTest.cpp

```diff
@@ -163,6 +210,11 @@ TEST_F(SparkCastExprTest, decimalToIntegral) {
   testDecimalToIntegralCasts<int8_t>();
 }
 
+TEST_F(SparkCastExprTest, decimalToFloat) {
+  testDecimalToFloatCasts<float>();
+  testDecimalToFloatCasts<double>();
```

Comment:
I took a try with above tests and found `testDecimalToFloatCasts<double>()` could work with the original code. The problem is in the float type, while I also find with a tiny fix we could make it work: use double kind for the tryCast conversion for float type, seeing https://github.com/facebookincubator/velox/commit/23959e53a09ef49b9eb6395bd126cc0e9ebeab67. In this way we could avoid introducing the Spark-like conversion which brings large performance regression as the benchmark shows.
> util::Converter<TypeKind::DOUBLE>::tryCast